source/use-cases/sensitive-data-encryption.txt

```diff
@@ -57,12 +57,222 @@ cubilia Curae; Nunc non interdum purus, ultricies laoreet tortor.
 B. Define a JSON Schema
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-Proin non mi a felis luctus vulputate nec pharetra odio. Ut pretium
-scelerisque nulla, sed euismod nisi pellentesque sit amet. Proin mollis
-mauris eu libero ullamcorper rutrum. Pellentesque vitae enim sed magna
-accumsan maximus et quis mi. Ut vel laoreet turpis. Vivamus vitae odio
-sed arcu finibus pretium vel maximus nisi. Nullam gravida consequat
-porta.
+MongoDB uses JSON Schema to validate data that is written to a collection.
+Within the context of Field Level Encryption, JSONSchema is used to
+specify several configuration details.
+For instance, you can specify the form of encryption you want to use  -
+deterministic or random.
+You can also specify the BSON Type of the field you are encrypting. In
+addition, you can 
+specify the key you want to use. This is particularly important, because
+it means that you can use different keys for different fields.
+
+The following example goes over the creation of a schema for a collection
+that represents patient data
+in a medical office. Some fields use deterministic encryption, and others
+use random (probabilistic) encryption. 
+
+The structure for the schema will be as follows:
+
+.. code::
+
+    fullName: String
+    address: Encrypted String using random encryption
+    ssn: Encrypted Int using deterministic encryption
+    last4SSN: Int
+    patientInfo: Object
+    patientInfo.phone Encrypted Int using deterministic encryption
+    patientInfo.provider String
+    medicalRecords: Encrypted array of documents using random encryption
+
+Explanation of encryption choices
+``````````````````````````````````
+* ``address`` is randomly encrypted because it is more secure for strings to be randomly encrypted to prevent hackers from decrypting by building a cipher-dictionary
+* ``ssn`` is deterministically encrypted because deterministic encryption provides filtering ability 
+* ``patientInfo.phone`` is deterministically encrypted for the same reason as ssn, and also it provides an example of encryption with a nested document
+* ``medicalRecords`` are randomly encrypted because entire arrays are only encryptable with random encryption
```

Comment:
We should introduce the (unencrypted) schema in the **Problem** section of the guide intro so that we have immediate context. As part of that, we should call out which fields we want to encrypt and _why_ we want to encrypt it.

These bullets mostly cover the _how_ of encryption - e.g. we randomly encrypt `address`, but why does `address` need to be private in the first place?

Reply:
I think these bullets cover the "why" rather than "how". These bullets are answering the question of "why are we choosing a certain form of encryption for a certain field", rather than "why are we encrypting". I think it makes more sense to answer why we are choosing to use one form of encryption over another. I also think it's the better question to answer. 

It's fair to assume our readers are smart enough to draw the conclusion that "address is encrypted because people might not want that info being in plain text", but we can't make the assumption that our readers know why address should be randomly encrypted instead of deterministically encrypted.

Reply:
Agreed that "why are we choosing a certain form of encryption for a certain field" is important.

I still think it's important that we talk about "why are we encrypting this field at all" because it's part of the logical narrative (we have this data, we want to protect these fields because reasons) and because it helps non-US users who might not understand what a social security number even is, let alone why I would encrypt the full thing but not the last four digits.

Reply:
Is there a use case for filtering an SSN? Full SSN is very private and nobody should really ever search based on that afaict.

Similarly, I can imagine a use case where I would want to filter based on address (e.g. "show me only MedcoMD patients that live in Cleveland.") which would require deterministic encryption.

Reply:
On second thought, you're right. SSN makes more sense as randomly encrypted because it doesn't require filtering, and address should be deterministically encrypted because they do require filtering. Fixed in the most recent commit.